code/__HELPERS/areas.dm

```diff
@@ -94,11 +94,6 @@ GLOBAL_LIST_INIT(typecache_powerfailure_safe_areas, typecacheof(/area/engineerin
 
 	newA.reg_in_areas_in_z()
 
-	var/list/firedoors = oldA.firedoors
-	for(var/door in firedoors)
-		var/obj/machinery/door/firedoor/FD = door
-		FD.CalculateAffectingAreas()
```

Comment:
Don't you still need this if areas can close all their firelocks?

Reply:
no, because the firelocks will add itself to one area, before it added itself to multiple areas since the breach/fire could be from one area going into another, but since they automatically shutdown there is no real need to close a different firedoor area from other areas

Reply:
Wouldn't this make pulling on fire alarms look a bit funky though?

Reply:
mm yeah some place might look weird with some firelock not closing, i'll modify this so that firedoors will automatically check for adjacent areas

Reply:
added to next commit